src/transformers/models/isaac/modular_isaac.py

```diff
@@ -0,0 +1,2109 @@
+            else:
+                scale_max = scale
+        scale = scale_min
+        target_height = get_scaled_image_size(scale, image_height, patch_size, pixel_shuffle_scale)
+        target_width = get_scaled_image_size(scale, image_width, patch_size, pixel_shuffle_scale)
+        return target_height, target_width
+
+
+def patchify_vision(image: torch.Tensor, patch_size: int) -> torch.Tensor:
+    r"""Convert normalized images into flattened ViT-style patches.
+
+    Args:
+        image (`torch.Tensor`):
+            Tensor of shape `(num_images, height, width, channels)`.
+        patch_size (`int`):
+            Edge length of the square patches
+
+    Returns:
+        `torch.Tensor`:
+            Patch tensor where each position stores the flattened pixels belonging to that patch.
+
+    Raises:
+        ValueError: If `height` or `width` is not divisible by `patch_size`.
+    """
+    num_images, height, width, channels = image.shape
+    if height % patch_size or width % patch_size:
+        raise ValueError(f"Dimensions of images {image.shape} are not divisible by patch_size={patch_size}.")
+    patches = image.reshape(num_images, height // patch_size, patch_size, width // patch_size, patch_size, channels)
+    patches = patches.permute(0, 1, 3, 2, 4, 5)
+    patches = patches.reshape(
+        num_images, height // patch_size, width // patch_size, channels * patch_size * patch_size
+    )
+    return patches
+
+
+class IsaacConfig(PretrainedConfig):
+    """Configuration class for Isaac multimodal model.
+
+    This configuration corresponds to checkpoints such as
+    [Perceptron/isaac-base](https://huggingface.co/Perceptron/isaac-base).
+    """
+
+    model_type = "isaac"
+    sub_configs = {"vision_config": IsaacVisionConfig, "text_config": Qwen3Config}
+    image_processor_type = "IsaacImageProcessor"
+
+    def __init__(
+        self,
+        vision_config: Optional[IsaacVisionConfig] = None,
+        text_config: Optional[Union[Qwen3Config, dict]] = None,
+        vision_rescale_factor: float = 1 / 255,
+        max_sequence_length: int = 16384,
+        vision_token: str = "<image>",
+        **kwargs,
+    ):
+        self._rope_parameters: Optional[dict[str, Any]] = None
+        attn_implementation = kwargs.get("attn_implementation")
+
+        if isinstance(text_config, dict):
+            self.text_config = self.sub_configs["text_config"](**text_config)
+        elif text_config is None:
+            self.text_config = self.sub_configs["text_config"]()
+
+        super().__init__(**kwargs)
+
+        if self._rope_scaling is None:
+            self._rope_scaling = getattr(self.text_config, "rope_scaling", None)
+        else:
+            self.text_config.rope_scaling = self._rope_scaling
+
+        # Keep rope parameters alias in sync with upstream expectations
+        self._rope_parameters = self._rope_scaling
+
+        # Mirror frequently accessed Qwen3 attributes at the composite config level for BC.
```

Comment:
there should be 0 BC comments, or code because the mode is new. Please remove this